runtime.md

```diff
@@ -90,18 +94,27 @@ Attempting to start an already started container MUST have no effect on the cont
 `run <container-id>`
 
 This operation MUST generate an error if it is not provided the container ID.
+Attempting to run a container that does not exist MUST generate an error.
 This operation MUST invoke the `create` operation, and if there are no errors, then invoke the `start` operation.
 The implementation MAY immediately execute `start` without any pause after `create` completes.
 For example if the implementation used a temporary process to implement `create`, it need not do this in the implementation of `run`.
 
+### Stop
+`stop <container-id>`
+
+This operation MUST generate an error if it is not provided the container ID.
+Attempting to stop a container that is not running MUST have no effect on the container and MUST generate an error.
+This operation MUST stop all of the processes running within the scope of the container.
+This operation MUST NOT delete any resources associated with the container, except for the PID namespace.
```

Comment:
I'm still not clear on why any of this requires hooks. Hooks already today run after the process and all namespaces have gone. Using the sandbox container approach (or bind mounts after create) you can arrange to have any subset of namespaces exist or not exist when a particular container exits.